cloudsync_onedrive.py

```diff
@@ -806,18 +831,12 @@ def events(self) -> Generator[Event, None, None]:      # pylint: disable=too-man
 
             for change in events:
                 event = self._convert_to_event(change, new_cursor)
-                log.debug("converted event %s as %s", change, event)
                 filter_result = self._filter_event(event)
                 if filter_result == EventFilter.IGNORE:
-                    if event:
-                        log.debug("ignore event: %s %s %s", event.path, event.oid, event.exists)
                     continue
-                if filter_result == EventFilter.WALK:
+                if filter_result == EventFilter.WALK and event.otype == DIRECTORY:
                     log.debug("directory created in or renamed into root - walking: %s", event.path)
-                    try:
-                        yield from self.walk_oid(event.oid)
-                    except CloudFileNotFoundError:
-                        pass
+                    yield from self._walk_filtered_directory(event.oid, walk_history)
```

Comment:
seems strange this wasn't hit by coverage?

Reply:
yeah, the new test only hits the actual walk function.

will add a separate one to hit this codepath in events()

Reply:
done